aiida/orm/utils/serialize.py

```diff
@@ -33,159 +36,91 @@
 _PLUMPY_BUNDLE = '!plumpy:bundle'
 
 
-def represent_enum(dumper, enum):
-    """Represent an arbitrary enum in yaml.
-
-    :param dumper: the dumper to use.
-    :type dumper: :class:`yaml.dumper.Dumper`
-    :param bundle: the bundle to represent
-    :return: the representation
-    """
+def represent_enum(dumper: yaml.Dumper, enum: Enum) -> yaml.ScalarNode:
+    """Represent an arbitrary enum in yaml."""
     loader = get_object_loader()
     return dumper.represent_scalar(_ENUM_TAG, f'{loader.identify_object(enum)}|{enum.value}')
 
 
-def enum_constructor(loader, serialized):
-    """Construct an enum from the serialized representation.
-
-    :param loader: the yaml loader.
-    :type loader: :class:`yaml.loader.Loader`
-    :param bundle: the enum representation.
-    :return: the enum.
-    """
-    deserialized = loader.construct_scalar(serialized)
+def enum_constructor(loader: yaml.Loader, serialized: yaml.Node) -> Enum:
+    """Construct an enum from the serialized representation."""
+    deserialized: str = loader.construct_scalar(serialized)  # type: ignore
```

Comment:
What are the warnings here for the `loader.construct_scalar`? If they cannot be fixed, should we at least maybe add the more specific error code to ignore instead of a blanket ignore?